### PR TITLE
Added extended Gradle support

### PR DESCRIPTION
**Pull request by lievendoclo**

### CODE REVIEW

File: lib/gradle/build/helpers.ts [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+import * as df from "dateformat";
+import { parseProperties } from "../../properties/propertiesParser";
+import { GradleProjectIdentifier } from "../parse/buildGradleParser";
+import { IsGradle } from "../pushtest/gradlePushTests";
+import { gradleCommand } from "./gradleBuilder";
+
+async function newVersion(sdmGoal: SdmGoalEvent, p: Project): Promise<string> {
+    const pi = await GradleProjectIdentifier(p);
+    const branch = sdmGoal.branch.split("/").join(".");
+    return `${pi.version}-${branch}.${df(new Date(), "yyyymmddHHMMss")}`;
```
Comment (cdupuis) [inline lib/gradle/build/helpers.ts]: should use `formatDate` function from sdm.

File: lib/gradle/build/helpers.ts [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+    return gradleCommand(p, {
+        progressLog: gi.progressLog,
+        tasks: ["build"],
+        args: [{name: "artifact.name", value: p.id.repo}],
+        errorFinder: SuccessIsReturn0ErrorFinder,
+    });
+}
+
+export const GradleBuild: GoalProjectListenerRegistration = {
+    name: "gradle-build",
```
Comment (cdupuis) [inline lib/gradle/build/helpers.ts]: remove `-`

File: lib/gradle/review/hasGradleVersion.ts [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+} from "@atomist/automation-client";
+import { ReviewerRegistration } from "@atomist/sdm";
+import { IsGradle } from "../pushtest/gradlePushTests";
+
+/**
+ * Checks whether a project has a gradle.properties and if so, if a version
+ * is defined in that properties file.
+ */
+export const HasVersionDefinedInGradleProperties: ReviewerRegistration = {
+    name: "gradle-version-check",
```
Comment (cdupuis) [inline lib/gradle/review/hasGradleVersion.ts]: Should be `GradleVersionInspection`

Comment (lievendoclo) [reply lib/gradle/review/hasGradleVersion.ts]: done

Comment (cdupuis) [reply lib/gradle/review/hasGradleVersion.ts]: And the name too.

File: lib/gradle/build/helpers.ts [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+} from "@atomist/sdm-core";
+import { parseProperties } from "../../properties/propertiesParser";
+import { GradleProjectIdentifier } from "../parse/buildGradleParser";
+import { IsGradle } from "../pushtest/gradlePushTests";
+import { gradleCommand } from "./gradleBuilder";
+
+async function newVersion(sdmGoal: SdmGoalEvent, p: Project): Promise<string> {
+    const pi = await GradleProjectIdentifier(p);
+    const branch = sdmGoal.branch.split("/").join(".");
+    return `${pi.version}-${branch}.${formatDate(new Date(), "yyyymmddHHMMss")}`;
```
Comment (cdupuis) [inline lib/gradle/build/helpers.ts]: could just be `formatDate()` as you provided all the defaults.

**Pull request merged**